api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -110,6 +114,20 @@ public void setUserProvider(Provider<User> userProvider) {
       fcCluster = this.notebooksService.createCluster(
           project, NotebooksService.DEFAULT_CLUSTER_NAME, userProvider.get().getEmail());
     }
+    if (org.pmiops.workbench.notebooks.model.ClusterStatus.ERROR.equals(fcCluster.getStatus())) {
+      User user = this.userProvider.get();
+      user.setClusterCreateRetries(user.getClusterCreateRetries() + 1);
+      this.userDao.save(user);
+      if (user.getClusterCreateRetries() <= 2) {
+        this.notebooksService.deleteCluster(project, NotebooksService.DEFAULT_CLUSTER_NAME);
```

Comment:
I think it would make sense to log a warning when we do this delete along with the retry count. If you're feeling ambitious, you could also log `cluster.getErrors()`, which might give some insight into what happened.